plugin/al/usdmayautils/AL/usdmaya/utils/MeshUtils.cpp

```diff
@@ -751,8 +759,171 @@ bool MeshImportContext::applyEdgeCreases()
+        continue;
+
+      MStatus s;
+  #if MAYA_API_VERSION >= 201800
+      colourSetName = fnMesh.createColorSetWithName(colourSetName, nullptr, nullptr, &s);
+  #else
+      colourSetName = fnMesh.createColorSetWithName(colourSetName, nullptr, &s);
+  #endif
+      if (s)
+      {
+        s = fnMesh.setCurrentColorSetName(colourSetName);
+        if (s)
+        {
+          // Prepare maya colours array
+          MFnMesh::MColorRepresentation representation;
+          if (vtValue.IsHolding<VtArray<GfVec3f> >())
+          {
+            //If we can find the special displayColorToken used by USD, let's check for the optional matching displayOpacityToken too
+            bool setCombinedDisplayAndOpacityColourSet = false;
+            if (name == displayColorToken)
+            {
+              UsdPrim prim = mesh.GetPrim();
+              if (prim.HasAttribute(primvarDisplayOpacityToken))
+              {
+                UsdAttribute usdAttr = prim.GetAttribute(primvarDisplayOpacityToken);
+                UsdGeomPrimvar primvar = UsdGeomPrimvar(usdAttr);
+                VtValue opacityValues;
+                if (primvar.Get(&opacityValues, m_timeCode))
+                {
+                  if (opacityValues.IsHolding<VtArray<float>>())
+                  {
+                    const VtArray<float> rawValOpacity = opacityValues.UncheckedGet<VtArray<float>>();
+                    colours.setLength(rawValOpacity.size());
+                    const VtArray<GfVec3f> rawValColour = vtValue.UncheckedGet<VtArray<GfVec3f> >();
+                    assert(rawValOpacity.size() == rawValColour.size());
+        
+                    for (uint32_t i = 0, n = rawValColour.size(); i < n; ++i)
+                      colours[i] = MColor(rawValColour[i][0], rawValColour[i][1], rawValColour[i][2], rawValOpacity[i]);
+                    representation = MFnMesh::kRGBA;
+                    setCombinedDisplayAndOpacityColourSet = true;
+                  }
+                }
+              }
+            }
+            if (!setCombinedDisplayAndOpacityColourSet) 
+            {
+              const VtArray<GfVec3f> rawVal = vtValue.UncheckedGet<VtArray<GfVec3f> >();
+              colours.setLength(rawVal.size());
+              for (uint32_t i = 0, n = rawVal.size(); i < n; ++i)
+                colours[i] = MColor(rawVal[i][0], rawVal[i][1], rawVal[i][2]);
+              representation = MFnMesh::kRGB;
+            }
+          }
+          else if (vtValue.IsHolding<VtArray<GfVec4f> >())
+          {
+            const VtArray<GfVec4f> rawVal = vtValue.UncheckedGet<VtArray<GfVec4f> >();
+            colours.setLength(rawVal.size());
+            memcpy(&colours[0], (const float*) rawVal.cdata(), sizeof(float) * 4 * rawVal.size());
+            representation = MFnMesh::kRGBA;
+          }
+        
+          // Set colors
+          if (!fnMesh.setColors(colours, &colourSetName, representation))
```

Comment:
error: ‘representation’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
`if (!fnMesh.setColors(colours, &colourSetName, representation))`